nil/tests/archive_node/suite_archive_node.go

```diff
@@ -0,0 +1,95 @@
+//go:build test
+
+package tests
+
+import (
+	"context"
+	"sync"
+
+	"github.com/NilFoundation/nil/nil/internal/db"
+	"github.com/NilFoundation/nil/nil/internal/network"
+	"github.com/NilFoundation/nil/nil/internal/types"
+	"github.com/NilFoundation/nil/nil/services/nilservice"
+	"github.com/NilFoundation/nil/nil/tests"
+)
+
+var (
+	oldProtocolVersion = "old"
+	newProtocolVersion = "new"
+)
+
+func protocolVersionOption(protocolVersion string) network.Option {
+	return func(c *network.Config) error {
+		c.ProtocolVersion = protocolVersion
+		return nil
+	}
+}
+
+type SuiteArchiveNode struct {
+	tests.ShardedSuite
+
+	cancel func()
+	wg     sync.WaitGroup
```

Comment:
`tests.ShardedSuite` has waitGroup. Do you need another one?

Reply:
This is the ported code. In general, this `WaitGroup` is responsible only for the archive node, and `ShardedSuite.wg` is responsible for the cluster. We need this separation at least in the restart test, when we stop and restart only the archive node.